site2/docs/cookbooks-message-throttling.md

```diff
@@ -0,0 +1,86 @@
+> by blocks of data.
+
+When delivering a message to the client, the Broker looks at the size and number of messages per data block, calculates
+an average and caches it in memory, using this average to estimate how many data blocks to be read. What if there is no
+estimated value in memory when the first reading of a new topic? The current workaround is that just delivers one data
+block to clients if it is the first delivery of a new topic. Of course, the broker does not store all the information
+of data blocks for a long time to process the average accurately, which would waste memory. Instead, a compromise
+algorithm 'avg = (history-avg * 0.9 + new-avg * 0.1)' is used.
+
+### Only three granularity supports
+- Throttling limit per broker: Often applied to overload protection of broker or storage. If only to prevent the
+  excessive read requests results in Storage can not work well, you can disabled the feature
+  `dispatchThrottlingOnNonBacklogConsumerEnabled` (default is enabled). After disabled this feature, if all consumers
+  in one subscription have no backlog(it is clear that almost all read requests can hit the cache), then message
+  dispatch throttling is turned off automatically，and if any consumer has backlog, it will be turned on automatically.
+- Throttling limit per topic: The maximum number of messages that can be delivered to clients per unit of time for the
+  same topic. If there are multiple partitions, this rule indicates the maximum number of messages each partition can
+  deliver per unit time. Total limiting of multiple partitions on the same topic is not supported. For example, Topic
+  `t2` has two partitions, and we set the limit of each topic to `10/s`, then the rate limit of `t2-partition-0` is
+  `10/s`, and the rate limit of `t2-partition-1` is also `10/s`. The speed limit for the whole `t2` is `20/s`. There
+  are three ways to set the rule of topics(Effective priority, from low to high):
+  - set by [configuration of broker](https://pulsar.apache.org/docs/reference-configuration/#broker); 
+  - set by [policies of namespace](https://pulsar.apache.org/docs/2.10.x/admin-api-namespaces/#configure-dispatch-throttling-for-topics);
+  - set by `pulsar-admin topicPolicies set-dispatch-rate`.
+- Throttling limit per subscription: The maximum number of messages that same subscription can deliver to clients per
+  unit time. If the subscribed topic has multiple partitions, this rule indicates the maximum number of messages the
+  subscription can deliver per partition per unit time. For example, topic `t2` has two partitions, we create a
+  subscription `sub1` to `t2`, and set rate limit to `10/s`. Then the speed limit for `sub1` in `t2-partition-0` is
+  `10/s`, and the speed limit for `sub1` in `t2-partition-1` is `10/s` too. So the speed limit for the whole `sub2` is
+  `20/s`. There are also three ways to set the rule of subscriptions(Effective priority, from low to high):
```

Comment:
It is unclear.

so, if we sum sub1 and sub2 limits, it will be 40/s?

Reply:
ah, the `sub2` is a typo. Thanks